chapters/what_neural_search.md

```diff
@@ -0,0 +1,103 @@
+
+In short, if data can be encoded to something a computer can understand, it can be searched with neural search.
+
+##### Flexibility
+
+Traditional searches are very brittle. For example, unless you create a rule that states "red", "scarlet" and "crimson" are very similar, there's no way the search engine would "know" that. So if a user searches `red sneakers`, crimson or scarlet sneakers wouldn't rank highly in the search results.
+
+Neural search uses algorithms that have been trained on huge datasets to understand natural language. So it's likely whatever synonym for red you can think up, the algorithm (and thus your search engine) already knows it. The result? Crimson and scarlet sneakers rank highly for the search term `red sneakers`.
+
+##### Forgiveness
+
+In a similar way, without a spell-checking library, traditional search can't understand that when a user types `pokeman` or `pokey man` they mean `pokemon`. And even with a spell-checking library, often new words (like Pokemon) just aren't in the dictionary.
+
+Neural search can make a "best guess" in situations like these. If a word exists in the training dataset (which are often more up-to-date than spell-checking dictionaries) it can guess a user's intent from their misspelled query.
+
+#### Comparing neural and symbolic search
+
+When we talk about "traditional search", we typically mean symbolic search. Symbolic search frameworks include Lucene, Solr, and Elastic Search. 
+
+Comparing neural and symbolic search is like comparing apples and oranges. Symbolic search excels in its given domain of single-language NLP, while neural search is a lot more flexible but still relatively new.
+
+While neural search has many advantages, there are still some situations where it makes sense to use traditional search:
+
+- Your search needs are constrained and unlikely to change over time
+- You are searching for text in just one language
+- Speed is more important than getting the most accurate results
+- You want to understand **why** you're getting your *those* results
+- Your computing power is highly limited
+- You have a small amount of data (neural search requires big data)
+
+On the other hand, there are some situations where neural search makes more sense
+
+- Your search needs are likely to expand over different languages and data types
+- Accurate results are important, even if you can't perfectly understand why you got *those* results
+- You have a "noisy" set of data to search through
+- You have a large dataset for the neural network to "learn"
```

Comment:
this is the only line I'd disagree with, with pre-trained models I think you can work with quite a small corpus. Even two or three hundred image + pre-trained image net would work.  What do you think?

Reply:
Yup. This section was adapted from my old version which was adapted from Han's original version. I think we've seen it do well in action so I'll remove that bullet point

Reply:
Resolved